### PR TITLE
Implement account creation flow

### PR DESCRIPTION
**Pull request by Vladyslav-Kuksiuk**
This PR adds an implementation of the account creation flow and adds the email address to the user.
Now email can be reserved and user registered only by account creation process.

Account creation flow model:
image

### CODE REVIEW

File: model/src/main/proto/spine_examples/chatspn/account/user_commands.proto [hunk 1]
```diff
@@ -44,4 +46,10 @@ message RegisterUser {
 
     // A name of the user to register.
     string name = 2 [(required) = true];
+
+    // An email of the user to register.
+    spine.net.EmailAddress email = 3 [(required) = true];
+
+    // The ID of the account creation process that has initiated the user registration.
+    AccountCreationId account_creation_process = 4 [(required) = true];
```
Comment (Artem-Semenov-dev) [inline model/src/main/proto/spine_examples/chatspn/account/user_commands.proto:54]: Maybe ```initiator``` is better here?

File: server/src/test/java/io/spine/examples/chatspn/server/account/AccountCreationTest.java [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+import io.spine.testing.core.given.GivenUserId;
+import io.spine.testing.server.blackbox.ContextAwareTest;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+
+import static io.spine.examples.chatspn.server.given.AccountCreationTestEnv.sendRandomCreateAccountCommand;
+import static io.spine.testing.TestValues.randomString;
+
+@DisplayName("`AccountCreation` should")
+class AccountCreationTest extends ContextAwareTest {
```
Comment (Artem-Semenov-dev) [inline server/src/test/java/io/spine/examples/chatspn/server/account/AccountCreationTest.java:50]: Looks `final` to me.

File: server/src/test/java/io/spine/examples/chatspn/server/account/AccountCreationTest.java [hunk 3]
```diff
@@ -0,0 +1,225 @@
...
+
+    @Override
+    protected BoundedContextBuilder contextBuilder() {
+        return ChatsContext.newBuilder();
+    }
+
+    @Test
+    @DisplayName("emit `AccountCreated` event if the process is finished successfully and archive itself")
```
[the remark this answers is not in the document]
Comment (Vladyslav-Kuksiuk) [reply server/src/test/java/io/spine/examples/chatspn/server/account/AccountCreationTest.java]: Fixed.

File: model/src/main/proto/spine_examples/chatspn/account/reserved_email_commands.proto [hunk 4]
```diff
@@ -45,4 +46,7 @@ message ReserveEmail {
 
     // The ID of the user who tells to reserve the email address.
     spine.core.UserId user = 2 [(required) = true];
+
+    // The ID of the account creation process that has initiated the email reservation.
+    AccountCreationId account_creation_process = 3 [(required) = true];
```
Comment (armiol) [inline model/src/main/proto/spine_examples/chatspn/account/reserved_email_commands.proto:51]: I would name this `process`, it's simple enough.

File: server/src/test/java/io/spine/examples/chatspn/server/account/AccountCreationTest.java [hunk 5]
```diff
@@ -0,0 +1,225 @@
...
+    @Override
+    protected BoundedContextBuilder contextBuilder() {
+        return ChatsContext.newBuilder();
+    }
+
+    @Test
+    @DisplayName("emit the `AccountCreated` event if the process is finished successfully and archive itself")
+    void createdEvent() {
+        CreateAccount command = sendRandomCreateAccountCommand(context());
+        AccountCreated expectedEvent = AccountCreated
```
Comment (armiol) [inline server/src/test/java/io/spine/examples/chatspn/server/account/AccountCreationTest.java:61]: Could you please start extracting some of code devoted to signal creation, to the respective test environment?

**Review by armiol — CHANGES_REQUESTED** [review]
@Vladyslav-Kuksiuk please see my comments.

**Review by armiol — APPROVED** [review]
LGTM.

**Pull request merged**